site/app/controllers/admin/UsersController.php

```diff
@@ -28,36 +31,106 @@ public function run() {
 
     /**
      * @Route("/{_semester}/{_course}/users", methods={"GET"})
+     * @Route("/api/{_semester}/{_course}/users", methods={"GET"})
+     * @return Response
      */
-    public function listStudents() {
+    public function getStudents() {
         $students = $this->core->getQueries()->getAllUsers();
-        $reg_sections = $this->core->getQueries()->getRegistrationSections();
-        $rot_sections = $this->core->getQueries()->getRotatingSections();
-        $use_database = $this->core->getAuthentication() instanceof DatabaseAuthentication;
-
-        $this->core->getOutput()->renderOutput(array('admin', 'Users'), 'listStudents', $students, $reg_sections, $rot_sections, $use_database);
-        $this->renderDownloadForm('user', $use_database);
+        //Assemble students into sections
+        $sorted_students = [];
+        $download_info = [];
+        foreach ($students as $student) {
+            $rot_sec = ($student->getRotatingSection() === null) ? 'NULL' : $student->getRotatingSection();
+            $reg_sec = ($student->getRegistrationSection() === null) ? 'NULL' : $student->getRegistrationSection();
+            $sorted_students[$reg_sec][] = $student;
+            switch ($student->getGroup()) {
+                case User::GROUP_INSTRUCTOR:
+                    $grp = 'Instructor';
+                    break;
+                case User::GROUP_FULL_ACCESS_GRADER:
+                    $grp = 'Full Access Grader (Grad TA)';
+                    break;
+                case User::GROUP_LIMITED_ACCESS_GRADER:
+                    $grp = 'Limited Access Grader (Mentor)';
+                    break;
+                default:
+                    $grp = 'Student';
+                    break;
+            }
+            $first_name = str_replace("'", "&#039;", $student->getDisplayedFirstName());
+            $last_name = str_replace("'", "&#039;", $student->getDisplayedLastName());
```

Comment:
Why are we encoding this here?

Reply:
Not sure why. Copied these lines from the old code. Tried to remove them and added a user with apostrophe in the name and the csv looked okay.